src/librustc_typeck/collect.rs

```diff
@@ -1496,7 +1532,12 @@ pub fn explicit_predicates_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
             &hir::WherePredicate::RegionPredicate(ref region_pred) => {
                 let r1 = AstConv::ast_region_to_region(&icx, &region_pred.lifetime, None);
                 for bound in &region_pred.bounds {
-                    let r2 = AstConv::ast_region_to_region(&icx, bound, None);
+                    let r2 = match bound {
+                        hir::ParamBound::Outlives(lt) => {
```

Comment:
Hmm, if it ends up being used like this, maybe `ParamBound` should be named `GenericBound`? `WherePredicate` could also potentially become simpler.